### PR TITLE
More Bank 03 Routines

### PR DESCRIPTION
**Pull request by Pokeglitch**
Still need to figure out constants for tiles and movements

### CODE REVIEW

File: data/predef_pointers.inc [hunk 1]
```diff
@@ -16,26 +16,26 @@ PredefPointers:: ; 1:62d3
...
-	add_predef Functioncde3
-	add_predef Functioncde3_2 ; 8
-	add_predef Functioncde3_3
-	add_predef Functioncdf9
-	add_predef Functionce10_2
-	add_predef Functioncd33
+	add_predef GetItemAmount
+	add_predef HealParty
+	add_predef AddBCD
+	dbw BANK(AddBCD), AddBCD ; 8
```
Comment (mid-kid) [inline data/predef_pointers.inc]: Use the `dba` macro

File: engine/events/field_moves.asm [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+.reset
+	xor a
+	ld [wFieldMoveScriptID], a
+	ret
+.next
+	ld a, [wFieldMoveScriptID]
+	ld hl, CutTable
+	jp CallJumptable
+	
+CutTable
```
Comment (mid-kid) [inline engine/events/field_moves.asm]: Missing semicolon. Why not make it local?

File: engine/events/field_moves.asm [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+MovePlayerIntoWater: ; 03:5185
+	call InitMovementBuffer
+	call .get_movement_direction
+	call AppendToMovementBuffer
+	ld a, $32
+	call AppendToMovementBuffer
+	ld a, 0
+	ld hl, wMovementBuffer
+	call LoadMovementDataPointer
+.get_movement_direction
```
Comment (mid-kid) [inline engine/events/field_moves.asm]: Missing newline.

File: engine/events/field_moves.asm [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+	call GetMapEnvironment
+	cp TOWN
+	jr z, .success
+	cp ROUTE
+	jr z, .success
+	ld a, SCRIPT_ID_03
+	ld [wFieldMoveScriptID], a
+	xor a
+	ret
+.success
```
Comment (mid-kid) [inline engine/events/field_moves.asm]: Missing newline.

File: home/music.asm [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	push hl
+	push de
+	push bc
+	push af
+	call GetMapMusic
+	ld a, [wMapMusic]
+	cp e
+	jr z, .dont_play
+	push de
+	ld de, $0000
```
Comment (mid-kid) [inline home/music.asm]: Just a 0.

File: engine/events/field_moves.asm [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+.loop
+	call .next
+	jr nc, .loop
+	ld [wFieldMoveSucceeded], a
+	ret
+.reset
+	xor a
+	ld [wFieldMoveScriptID], a
+	ret
+.next
```
Comment (luckytyphlosion) [inline engine/events/field_moves.asm]: `.reset` and `.next` should be `.PascalCase`

**Comment by Pokeglitch** [comment]
Added in requested changes, and macros for the scripting and queueing

**Pull request merged**